xaitk_saliency/utils/masking.py

```diff
@@ -28,3 +29,36 @@ def generate_masked_images(
         masked_img = np.multiply(mask, img, casting='unsafe')
         masked_imgs.append(PIL.Image.fromarray(np.uint8(masked_img)))
     return masked_imgs
+
+
+def weight_regions_by_scalar(
+    scalar_vec: np.ndarray,
+    masks: np.ndarray
+) -> np.ndarray:
+    """
+    Weight some binary masks reguib with its respective vector in scalar_vec.
+
+    We expect the "masks" matrices and the image to be the same height and
+    width, and be valued in the [0, 1] floating-point range. The length
+    and order of scalar_vec and masks are assumed to be the same.
+    In the mask matrix, higher values correspond to regions of the image that
+    preserved. E.g. a 0 in the mask will translate to blacking out the
+    corresponding location in the source image.
+
+    :param scalar_vec: Weights for image regions in masks.
+    :param masks: Mask array in the [nMasks, Height, Weight] shape format.
+
+    :return: A numpy array representing the weighted heatmap.
+    """
+
+    # Creating an empty heatmap for aggregating weighted matrices
+    heatmap = np.zeros((masks.shape[-2:]))
+
+    # Iterating through perturbation mask and respective weight
+    for weight, prtb_region in zip(scalar_vec, masks):
+        heatmap += ((1 - prtb_region) * weight)
+
+    normalized_heatmap = heatmap / len(scalar_vec)
+    final_heatmap = minmax_scale(normalized_heatmap.ravel(),
+                                 feature_range=(0, 1)).reshape(heatmap.shape)
```

Comment:
I'm not sure saliency map normalization is in the scope of the computation or not? We may just want to return "raw" saliency maps to the user, and have them perform normalization as they see fit themselves. I guess that is open for discussion and will also apply to the other saliency implementations.

Reply:
That makes sense to me. I moved the scaling into the notebook and modified the weight_regions_by_scalar() to return raw saliency maps.

Reply:
We defined in the API document that saliency maps return in the [0,1] range. My potential concern with not bounding the output to something is that tools around saliency generator output, when configured with a different implementation, will not be able to handle the variety of unconstrained output.

Reply:
In running the example notebook, I can see that the output map is in the state of being not normalized. Did we come to more firm opinions on how this output should take form?

Reply:
Although normalization is more of a visualization thing and not related to generating process itself, @Purg's point about future conflicts between different versions is valid. The only drawback to always normalizing is a fixed dynamic range of values for better or worse depending upon the implementation, which is not a huge disadvantage from my perspective. @brianhhu let me know what you think. I can revert this change to add the normalization to OcclusionScoring.generate() instead of the notebook.